src/list/util/listUtils.ts

```diff
@@ -0,0 +1,27 @@
+import {v4 as uuidv4} from "uuid";
+
+import {ListProps} from "../List";
+
+function generateListItemKey<Item = any>({
+  listItemKeyGenerator,
+  listItemTestId,
+  item
+}: {
+  listItemKeyGenerator: ListProps["listItemKeyGenerator"];
+  listItemTestId: string;
+  item: Item;
+}) {
+  let key = uuidv4();
```

Comment:
Can we move `uuidv4` to the `else` block of the below if? `uuidv4` should be our fallback solution, currently, it's creating id for every list item, then removes it (if any other id exists such as listItemKeyGenerator or item.id )

Reply:
Yes, it's a good point! Thank you 🙏🏻